pkg/option/config.go

```diff
@@ -15,71 +15,123 @@
 package option
 
 import (
-	"fmt"
-	"strings"
+	"net"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/lock"
 )
 
-// Validator returns a validated string along with a possible error.
-type Validator func(val string) (string, error)
+const (
+	// AllowLocalhostAuto defaults to policy except when running in
+	// Kubernetes where it then defaults to "always"
+	AllowLocalhostAuto = "auto"
 
-// MapOptions holds a map of values and a validation function.
-type MapOptions struct {
-	vals      map[string]string
-	validator Validator
-}
+	// AllowLocalhostAlways always allows the local stack to reach local
+	// endpoints
+	AllowLocalhostAlways = "always"
+
+	// AllowLocalhostPolicy requires a policy rule to allow the local stack
+	// to reach particular endpoints or policy enforcement must be
+	// disabled.
+	AllowLocalhostPolicy = "policy"
+
+	// ModePreFilterNative for loading progs with xdpdrv
+	ModePreFilterNative = "native"
+
+	// ModePreFilterGeneric for loading progs with xdpgeneric
+	ModePreFilterGeneric = "generic"
+)
+
+// Config is the configuration used by Daemon.
+type Config struct {
```

Comment:
Since this type is now global, it should be called DaemonConfig or something similar. Previously nothing would import the daemon package (because it was main) so it didn't matter if it was exported or not, and the name had more context.